pkg/features/feature.go

```diff
@@ -0,0 +1,37 @@
+package features
+
+import (
+	"k8s.io/component-base/featuregate"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// Every feature gate should add method here following this template:
+	//
+	// // owner: @username
+	// // alpha: v1.X
+	// MyFeature featuregate.Feature = "MyFeature"
+
+	// ClusterClaim will manage the cluster-claim resources in the managed cluster. Disabling the
+	// the feature-gate can make the spoke-agent run without installing cluster-claim CRDs in the
```

Comment:
The cluster claim CRD is installed even when the feature gate is disabled unless we put the feature gate into Klusterlet CR and let registration-operator decide if cluster claim CRD should be installed.

Reply:
for clarification, the point of the feature-gate is not for chopping cluster-claim off intentionally from a managed-cluster. it's to make the agent work when the CRD cannot be installed due to insufficient permission. in our case, the crux is that neither registration-operator nor OCM admin doesn't have the cluster-scoped access to the managed cluster --- it doesn't actually have a choice between installation CRDs or not.

Reply:
My memory of the controller is that it creates `ManagedClusterClaim`, not a CRD.  Is this about the ManagedClusterClaim CR or the ManagedClusterClaim CRD?

Reply:
> My memory of the controller is that it creates `ManagedClusterClaim`, not a CRD. Is this about the ManagedClusterClaim CR or the ManagedClusterClaim CRD?

Related: can we update the comment to indicate
1. the end-user-impact of this being disabled
2. how an end-user knows this is disabled
3. how a deployer can mitigate this impact

Reply:
> Is this about the ManagedClusterClaim CR or the ManagedClusterClaim CRD?

i believe @elgnay is talking about the CRD. 

> My memory of the controller is that it creates ManagedClusterClaim, not a CRD. 

as far as i can see from the code, the cluster-claim controller doesnt actually create the cluster-claim, it just bounds the # of custom cluster-claims in the managed cluster by setting `--max-custom-cluster-claims` of the agent --- i'm a bit confused at which component is creating the claim? the intention of ManagedClusterClaim CRD is even unclear to me either. 🤔

> Related: can we update the comment to indicate

just enrich the doc for the feature-gate, PTAL

Reply:
@yue9944882 You can find the intention of ManagedClusterClaim from this [enhancement proposal](https://github.com/open-cluster-management/enhancements/tree/master/enhancements/sig-architecture/4-cluster-claims). Registration agent focus on vendor-neutral logic. It only collects all claims on managed cluster and puts them into status of managed cluster on hub. The logic of cluster claim generation might be vendor/product specific. It now been done by `work-manager`, an addon running on managed cluster. Of course, claims could be created by any other agent as well. So as a summary, 
- ManagedClusterClaim CRD is created by registration operator;
- and ManagedClusterClaim CRs are created by work-manager addon;
- Registration agent watchs ManagedClusterClaim CRs and sync them to ManagedCluster on hub;

Reply:
> ManagedClusterClaim CRD is installed by klusterlet (https://github.com/open-cluster-management/registration-operator);

isnt registration-operator an completely optional component? e.g. prometheus-operator is optional for automating deployment for prometheus. the automation for installing ManagedClusterClaim CRD shouldnt be an blocking issue for introducing the feature-gate.

Reply:
> isnt registration-operator an completely optional component? 

Yes, it's optional for running registration agent. You are able to deploy it in your own way.